### ISSUE TITLE
Build Failure: TypeScript Errors in @copilotkit/react-textarea Regarding DialogPortalProps

### ISSUE THREAD

**Issue by thormuller** [CopilotKit/CopilotKit#334]
**Describe the bug**
When attempting to build the @copilotkit/react-textarea package, TypeScript compilation fails with errors related to the DialogPortalProps. Specifically, the className property is reported as missing from DialogPortalProps, which causes the build process to fail.

**To Reproduce**
- Ensure Node.js version pnpm version 7.15.0 is installed.
- Navigate to the root directory of the CopilotKit project.
- Run the command pnpm -w freshbuild.

**Expected behavior**
The build process completes without errors, and all packages compile successfully.

**Actual Behavior**
The build process fails with the following TypeScript errors in @copilotkit/react-textarea:

error TS2339: Property 'className' does not exist on type 'DialogPortalProps'.
error TS2322: Type '{ children?: ReactNode; container?: HTMLElement | null | undefined; forceMount?: true | undefined; className: string; }' is not assignable to type 'IntrinsicAttributes & DialogPortalProps'.
These errors indicate that the className property is either incorrectly used in the component or missing from the DialogPortalProps type definition.

**CONSOLE EXCERPT**
```@copilotkit/react-textarea:build: ESM ⚡️ Build success in 576ms
@copilotkit/react-ui:build: DTS ⚡️ Build success in 3570ms
@copilotkit/react-ui:build: DTS dist/index.d.ts                              599.00 B
@copilotkit/react-ui:build: DTS dist/components/chat/index.d.ts              438.00 B
@copilotkit/react-ui:build: DTS dist/context/index.d.ts                      12.00 B
@copilotkit/react-ui:build: DTS dist/hooks/use-copy-to-clipboard.d.ts        266.00 B
@copilotkit/react-ui:build: DTS dist/lib/utils.d.ts                          263.00 B
@copilotkit/react-ui:build: DTS dist/types/types.d.ts                        351.00 B
@copilotkit/react-ui:build: DTS dist/components/chat/Button.d.ts             237.00 B
@copilotkit/react-ui:build: DTS dist/components/chat/CodeBlock.d.ts          392.00 B
@copilotkit/react-ui:build: DTS dist/components/chat/Header.d.ts             231.00 B
@copilotkit/react-ui:build: DTS dist/components/chat/Icons.d.ts              1.09 KB
@copilotkit/react-ui:build: DTS dist/components/chat/Input.d.ts              249.00 B
@copilotkit/react-ui:build: DTS dist/components/chat/Markdown.d.ts           213.00 B
@copilotkit/react-ui:build: DTS dist/components/chat/Messages.d.ts           262.00 B
@copilotkit/react-ui:build: DTS dist/components/chat/Response.d.ts           275.00 B
@copilotkit/react-ui:build: DTS dist/components/chat/Suggestion.d.ts         780.00 B
@copilotkit/react-ui:build: DTS dist/components/chat/Textarea.d.ts           490.00 B
@copilotkit/react-ui:build: DTS dist/components/chat/Window.d.ts             297.00 B
@copilotkit/react-ui:build: DTS dist/components/chat/Popup.d.ts              379.00 B
@copilotkit/react-ui:build: DTS dist/components/chat/Sidebar.d.ts            383.00 B
@copilotkit/react-ui:build: DTS dist/components/chat/Chat.d.ts               3.85 KB
@copilotkit/react-ui:build: DTS dist/components/chat/ChatContext.d.ts        3.27 KB
@copilotkit/react-ui:build: DTS dist/types/suggestions.d.ts                  721.00 B
@copilotkit/react-ui:build: DTS dist/components/chat/props.d.ts              995.00 B
@copilotkit/react-ui:build: DTS dist/hooks/use-copilot-chat-suggestions.d.ts 1.97 KB
@copilotkit/react-ui:build: DTS dist/components/index.d.ts                   460.00 B
@copilotkit/react-ui:build: DTS dist/types/index.d.ts                        12.00 B
@copilotkit/react-ui:build: DTS dist/lib/utils.test.d.ts                     12.00 B
@copilotkit/react-ui:build: DTS dist/hooks/index.d.ts                        113.00 B
@copilotkit/react-textarea:build: src/components/ui/dialog.tsx(11,25): error TS2339: Property 'className' does not exist on type 'DialogPortalProps'.
@copilotkit/react-textarea:build: src/components/ui/dialog.tsx(12,27): error TS2322: Type '{ children?: ReactNode; container?: HTMLElement | null | undefined; forceMount?: true | undefined; className: string; }' is not assignable to type 'IntrinsicAttributes & DialogPortalProps'.
@copilotkit/react-textarea:build:   Property 'className' does not exist on type 'IntrinsicAttributes & DialogPortalProps'.
@copilotkit/react-textarea:build: 
@copilotkit/react-textarea:build: Error: error occured in dts build
@copilotkit/react-textarea:build:     at Worker.<anonymous> (/Users/thormuller/Documents/copilotkit/CopilotKit/CopilotKit/node_modules/.pnpm/tsup@6.7.0_znczopou26gt5k4c3ewblt7dyq/node_modules/tsup/dist/index.js:2281:26)
@copilotkit/react-textarea:build:     at Worker.emit (node:events:513:28)
@copilotkit/react-textarea:build:     at MessagePort.<anonymous> (node:internal/worker:247:53)
@copilotkit/react-textarea:build:     at [nodejs.internal.kHybridDispatch] (node:internal/event_target:761:20)
@copilotkit/react-textarea:build:     at exports.emitMessage (node:internal/per_context/messageport:23:28)
@copilotkit/react-textarea:build: DTS Build error
@copilotkit/react-textarea:build: RollupError: Failed to compile. Check the logs above.
@copilotkit/react-textarea:build:     at error (/Users/thormuller/Documents/copilotkit/CopilotKit/CopilotKit/node_modules/.pnpm/rollup@3.29.4/node_modules/rollup/dist/shared/rollup.js:353:30)
@copilotkit/react-textarea:build:     at Object.error (/Users/thormuller/Documents/copilotkit/CopilotKit/CopilotKit/node_modules/.pnpm/rollup@3.29.4/node_modules/rollup/dist/shared/rollup.js:1721:20)
@copilotkit/react-textarea:build:     at Object.error (/Users/thormuller/Documents/copilotkit/CopilotKit/CopilotKit/node_modules/.pnpm/rollup@3.29.4/node_modules/rollup/dist/shared/rollup.js:25627:42)
@copilotkit/react-textarea:build:     at generateDtsFromTs (/Users/thormuller/Documents/copilotkit/CopilotKit/CopilotKit/node_modules/.pnpm/tsup@6.7.0_znczopou26gt5k4c3ewblt7dyq/node_modules/tsup/dist/rollup.js:7498:22)
@copilotkit/react-textarea:build:     at /Users/thormuller/Documents/copilotkit/CopilotKit/CopilotKit/node_modules/.pnpm/tsup@6.7.0_znczopou26gt5k4c3ewblt7dyq/node_modules/tsup/dist/rollup.js:7505:59
@copilotkit/react-textarea:build:     at _nullishCoalesce (/Users/thormuller/Documents/copilotkit/CopilotKit/CopilotKit/node_modules/.pnpm/tsup@6.7.0_znczopou26gt5k4c3ewblt7dyq/node_modules/tsup/dist/rollup.js:1:198)
@copilotkit/react-textarea:build:     at Object.transform (/Users/thormuller/Documents/copilotkit/CopilotKit/CopilotKit/node_modules/.pnpm/tsup@6.7.0_znczopou26gt5k4c3ewblt7dyq/node_modules/tsup/dist/rollup.js:7505:18)
@copilotkit/react-textarea:build:     at /Users/thormuller/Documents/copilotkit/CopilotKit/CopilotKit/node_modules/.pnpm/rollup@3.29.4/node_modules/rollup/dist/shared/rollup.js:1914:40
@copilotkit/react-textarea:build:  ELIFECYCLE  Command failed with exit code 1.
@copilotkit/react-textarea:build: ERROR: command finished with error: command (/Users/thormuller/Documents/copilotkit/CopilotKit/CopilotKit/packages/react-textarea) /Users/thormuller/.nvm/versions/node/v19.9.0/bin/pnpm run build exited (1)
@copilotkit/react-textarea#build: command (/Users/thormuller/Documents/copilotkit/CopilotKit/CopilotKit/packages/react-textarea) /Users/thormuller/.nvm/versions/node/v19.9.0/bin/pnpm run build exited (1)

 Tasks:    5 successful, 6 total
Cached:    0 cached, 6 total
  Time:    25.157s 
Failed:    @copilotkit/react-textarea#build

 ERROR  run failed: command  exited (1)
 ELIFECYCLE  Command failed with exit code 1.
 ELIFECYCLE  Command failed with exit code 1.```

**Additional context**
Operating System: macOS 14.4.1
Node: 19.9.0

**Comment by ataibarkai** [comment]
Thanks for this report @thormuller !

As you mentioned on the Discord this sounds similar to https://github.com/CopilotKit/CopilotKit/issues/219 -
in both cases we can't reproduce this on our machines.

Thanks for listing the macOS / Node versions.
Any other setup that might be unique? I said this on the discord, can you please try with pnpm version 8.14.3?

**Comment by thormuller** [comment]
This issue is resolved by installing pnpm v8.14.3 and following the rest of the instructions. This version should be updated in the package.json because it currently specifies 7.15.0.
